10_rest/runnerdave/cmd/puppy_server/main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	puppy "github.com/anz-bank/go-course/10_rest/runnerdave/pkg/puppy"
+	store "github.com/anz-bank/go-course/10_rest/runnerdave/pkg/puppy/store"
+
+	"github.com/go-chi/chi"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	args           = os.Args[1:]
+	data           = kingpin.Flag("data", "data file").Short('d').Default("puppydata/data.json").ExistingFile()
+	port           = kingpin.Flag("port", "PORT").Short('p').Default("8080").String()
+	db             = kingpin.Flag("store", "STORE").Short('s').Default("sync").String()
+	out  io.Writer = os.Stdout
+)
+
+func main() {
+	_, err := kingpin.CommandLine.Parse(args)
+	if err != nil {
+		fmt.Fprintf(out, "Command line could not be parsed, error: %v", err)
+		return
+	}
+
+	d, _ := ioutil.ReadFile(*data)
+	puppies, uerr := unmarshalPuppies(d)
+	if uerr != nil {
+		return
```

Comment:
log / print error e.g. `log.Fatalf("cannot unmarshal puppies %v", err)` or `panic(err)`

Reply:
done

Reply:
nit: log.Fatal exits the program (with exit(1) that is), so you don't need the extra return line.